src/conformist.ml

```diff
@@ -1,330 +1,18 @@
-        (fun res el ->
-          match res, field.validator el with
-          | None, None -> None
-          | None, Some msg -> Some msg
-          | Some msg, _ -> Some msg)
-        None
-        l
-    in
-    let encoder (a : 'a List.t) = List.map field.encoder a |> List.concat in
-    make field.name meta decoder encoder default field.type_ validator true
-  ;;
-
-  let make_bool ?default ?meta ?(msg = "Invalid value provided") name =
-    let decoder input =
-      try Ok (bool_of_string (List.hd input)) with
-      | _ -> Error msg
-    in
-    let encoder input = List.[ string_of_bool input ] in
-    make name meta decoder encoder default "bool" always_valid false
-  ;;
-
-  let make_float
-      ?default
-      ?meta
-      ?(msg = "Invalid number provided")
-      ?(validator = always_valid)
-      name
-    =
-    let decoder string =
-      try Ok (float_of_string (List.hd string)) with
-      | _ -> Error msg
-    in
-    let encoder input = List.[ string_of_float input ] in
-    make name meta decoder encoder default "float" validator false
-  ;;
-
-  let make_int
-      ?default
-      ?meta
-      ?(msg = "Invalid number provided")
-      ?(validator = always_valid)
-      name
-    =
-    let decoder string =
-      try Ok (int_of_string (List.hd string)) with
-      | _ -> Error msg
-    in
-    let encoder input = List.[ string_of_int input ] in
-    make name meta decoder encoder default "int" validator false
-  ;;
-
-  let make_string ?default ?meta ?(validator = always_valid) name =
-    let decoder input =
-      try Ok (List.hd input) with
-      | _ -> Error "Invalid input provided"
-    in
-    let encoder id = List.[ id ] in
-    make name meta decoder encoder default "string" validator false
-  ;;
-
-  let make_date
-      ?default
-      ?meta
-      ?(msg = "Invalid date provided")
-      ?(validator = always_valid)
-      name
-    =
-    let decoder input =
-      try
-        match String.split_on_char '-' (List.hd input) with
-        | [ y; m; d ] ->
-          (match
-             int_of_string_opt y, int_of_string_opt m, int_of_string_opt d
-           with
-          | Some y, Some m, Some d -> Ok (y, m, d)
-          | _ -> Error msg)
-        | _ -> Error msg
-      with
-      | _ -> Error msg
-    in
-    let encoder (y, m, d) = List.[ Format.sprintf "%d-%d-%d" y m d ] in
-    make name meta decoder encoder default "date" validator false
-  ;;
-
-  let make_datetime
-      ?default
-      ?meta
-      ?(msg = "Invalid datetime provided")
-      ?(validator = always_valid)
-      name
-    =
-    let decoder string =
-      try
-        match Ptime.of_rfc3339 (List.hd string) with
-        | Ok (timestamp, _, _) -> Ok timestamp
-        | Error (`RFC3339 (_, _)) -> Error msg
-      with
-      | _ -> Error msg
-    in
-    let encoder ptime = List.[ Ptime.to_rfc3339 ptime ] in
-    make name meta decoder encoder default "time" validator false
-  ;;
+module String_error : Core.ERROR = struct
```

Comment:
@joseferben  when trying to pin this version, I get the following error message during the installation process

```
#    val validate :
#            'a any_field -> string list/2 -> String_error.error option
#        is not included in
#          val validate : 'a any_field -> string List.t -> error_msg option
#        File "src/core.ml", line 39, characters 4-68: Expected declaration
#        File "src/core.ml", line 344, characters 8-16: Actual declaration
#        File "src/core.ml", lines 333-337, characters 4-40:
#          Definition of type list/1
#        File "_none_", line 1:
#          Definition of type list/2
```

Reply:
@mabiede Yes I couldn't get it to compile yet, just drafted the API :)

Reply:
@mabiede Compilation errors fixed, in `test.ml` you find an example on how to apply the custom error module. What do you think?